service/application/src/test/java/com/solace/maas/ep/event/management/agent/subscriber/ScanJobPersistentMessageHandlerTests.java

```diff
@@ -0,0 +1,186 @@
+        // the waitForScanCompletion method will throw an exception after the timeout
+        when(scanManager.isScanComplete("scanId")).thenReturn(false);
+        solacePersistentMessageHandler.onMessage(inboundMessage);
+        // sleep for a while to allow the scan complete poll interval to pass
+        waitForScanCompletePolling(5);
+
+
+        // the scan command message processor should be called once by the persistent message handler
+        verify(scanCommandMessageProcessor, times(1)).processMessage(any());
+        // if the EMA is managed, the waitForScanCompletion method should be called
+        verify(scanCommandMessageProcessor, atLeastOnce()).waitForScanCompletion(any());
+        // the message should be acked after the scan is complete, even though it is timed out
+        verify(solacePersistentMessageHandler.getPersistentMessageReceiver(), times(1)).ack(inboundMessage);
+        List<ILoggingEvent> logs = listAppendersCanCommandMessageProcessor.list;
+        assertThat(logs.get(logs.size() - 1).getFormattedMessage())
+                .isEqualTo("Scan with id scanId did not complete within the expected time");
+    }
+
+    @Test
+    void testPersistentMessageHandlerScanCommandExceptionThrownMsgAcked() {
+        ScanCommandMessage scanCommandMessage =
+                new ScanCommandMessage("messagingServiceId",
+                        "scanId", List.of(SOLACE_ALL), List.of(EVENT_PORTAL));
+        when(inboundMessage.getPayloadAsString()).thenReturn(jsonString(scanCommandMessage));
+        when(inboundMessage.getProperty(MOPConstants.MOP_MSG_META_DECODER)).thenReturn(
+                ScanCommandMessage.class.getCanonicalName()
+        );
+        when(scanManager.scan(any())).thenThrow(new RuntimeException("Test exception thrown on purpose"));
+        solacePersistentMessageHandler.onMessage(inboundMessage);
+        // sleep for a while to allow the scan complete poll interval to pass
+        waitForScanCompletePolling(2);
+        List<ILoggingEvent> logs = listAppenderPersistentMessageHandler.list;
+        assertThat(logs.get(logs.size() - 1).getFormattedMessage())
+                .isEqualTo("Error while processing inbound message from queue for mopMessageSubclass: " + ScanCommandMessage.class.getCanonicalName());
+
+        // the scan command message processor should be called once by the persistent message handler
+        verify(scanCommandMessageProcessor, times(1)).processMessage(any());
+        verify(scanCommandMessageProcessor, atLeastOnce()).onFailure(any(), any());
+        // the message should be acked after the scan is complete
+        verify(solacePersistentMessageHandler.getPersistentMessageReceiver(), times(1)).ack(inboundMessage);
+    }
+
+    private void waitForScanCompletePolling(Integer additionalSeconds) {
+        try {
+            int timeout =eventPortalProperties.getWaitAckScanCompleteTimeoutSec() + additionalSeconds;
+            Thread.sleep(timeout* 1000L );
```

Comment:
This test takes a loooong time! can we set the `WaitAckScanCompleteTimeoutSec` to a smaller number than 300 seconds?